zerver/tests/test_rocketchat_importer.py

```diff
@@ -958,6 +958,25 @@ def test_do_convert_data(self) -> None:
         exported_usermessage_messages = self.get_set(messages["zerver_usermessage"], "message")
         self.assertEqual(exported_usermessage_messages, exported_messages_id)
 
+        # Tests for starred messages and wildcard mentions
+        usermessage_map: Dict[int, Dict[int, ZerverFieldsT]] = {}
+        for usermessage in messages["zerver_usermessage"]:
+            message_id = usermessage["message"]
+            user_id = usermessage["user_profile"]
+            if message_id not in usermessage_map:
+                usermessage_map[message_id] = {}
+            usermessage_map[message_id][user_id] = usermessage
```

Comment:
@timabbott I tried both ways (`parallel=1` and `parallel=1 --reverse`) and the tests are running fine on my local system in both the cases.

I think the issue here is that the test is skipping the for loop above (maybe to run non-deterministically?) and directly moving over to the tests below. So, is there any way to run the tests written below only after the for loop above has completed its execution?

Reply:
Individual tests don't run in parallel or multithreaded at all -- the parallelism is entirely at the level of running individual tests in different independent processes (and I'm not certain but possibly we only distribute tests by test class, so that all tests within a class run in series within the same process).  So I think we need a different theory for what's happening here.